src/util.rs

```diff
@@ -252,12 +225,47 @@ pub(crate) mod ptr {
             // details.
             #[allow(unstable_name_collisions)]
             match self._try_cast_into(_CastType::_Prefix) {
-                Some((slf, split_at)) if split_at == self._bytes_len => Some(slf),
+                Some((slf, split_at)) if split_at == self._len() => Some(slf),
                 Some(_) | None => None,
             }
         }
     }
 
+    impl<'a, T> Ptr<'a, [T]> {
+        /// The number of slice elements referenced by `self`.
+        fn _len(&self) -> usize {
```

Comment:
I'd like to suggest that we start getting into the practice of using `allow(dead_code)` at definition sites rather than `_name` at use-sites. It's a major chore to clean those use-sites up, especially when they appear in macros and comments and cannot be renamed automatically.

Reply:
Discussed offline: we'll transition to this in a separate PR.

Reply:
Uploaded a PR for this: https://github.com/google/zerocopy/pull/646